src/storage/core.rs

```diff
@@ -203,7 +203,22 @@ impl<K> Storage<K> {
             .active_blob
             .as_mut()
             .ok_or_else(Error::active_blob_not_set)?;
-        blob.write(record).await
+        match blob.write(record).await {
```

Comment:
It is wrong to downcast inside pearl. We can return the error we want from blob.write().

Reply:
hm, shouldn't this error then be in bob? or we should define this error in pearl but use in bob?

Reply:
And there is one more question: if we don't downcast this error in pearl, then this error will be received in disk_controller (according to todays version of bob) and in there we can organize processing of os error without converting to WorkDirUnavailable (Maybe I misunderstood you, waiting for an answer)